recipes/jupyterlab-lsp/meta.yaml

```diff
@@ -0,0 +1,116 @@
+{% set name = "jupyter-lsp" %}
+{% set ext_name = "@krassowski/jupyterlab-lsp" %}
+{% set version = "0.8.0" %}
+{% set build_number = "0" %}
+
+{% set py_spec = ">=3.5" %}
+{% set lab_spec = ">=2,<3.0.0a0" %}
+{% set ext_spec = "@^1.0.0" %}
+
+# List of jupyter_lsp py.tests ran for each known servers
+{% set server_tests = ["serverextension", "start_known", "listeners"] %}
+
+package:
+  name: {{ name }}-meta  # To avoid issue https://github.com/conda/conda-build/issues/3933
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 3abe20ec48a3f6b132653abe597af2b6cd0005539ab0d74e0ba203927124d333
+
+build:
+  noarch: python
+  number: {{ build_number }}
+
+outputs:
+  - name: {{ name }}
+    build:
+      number: {{ build_number }}
+      noarch: python
+    script: build_base.bat  # [win]
+    script: build_base.sh  # [not win]
+    requirements:
+      host:
+        - pip
+        - python {{ py_spec }}
+      run:
+        - notebook >=4.3.1
+        - python {{ py_spec }}
+        - setuptools
+    test:
+      requires:
+        - jupyterlab {{ lab_spec }}
+        - jupyterlab_server>=1.1.0
+        - nodejs >=10
+        - pip
+      commands:
+        - python -m pip check
+        - jupyter serverextension list 1>serverextensions 2>&1
+        - cat serverextensions | grep "{{ name|replace("-", "_") }}.*OK"  # [unix]
+        - jupyter labextension install {{ ext_name }}{{ ext_spec }}
+        - jupyter labextension list 1>labextensions 2>&1
+        - cat labextensions | grep "{{ ext_name }}.*OK"  # [unix]
+        - python -m jupyterlab.browser_check  # [not linux]
+  - name: {{ name }}-python
+    build:
+      number: {{ build_number }}
+      noarch: generic
+    requirements:
+      run:
+        - {{ pin_subpackage(name, exact=True) }}
+        # this combo appease pip-check, a canary for broken entry_points
+        # (used extensively by python-language-server and jupyter-lsp)
+        # in the future, this should hopefully be mostly handled by upstreams
+        - python-language-server >=0.31.9
+        - wrapt >=1.11,<=1.12.0a0
```

Comment:
Which exact upstream? A link to an issue would help in the future.

Reply:
@bollwyvl I blindly use your settings on that one:
 
> Here's the stuff I'm working on.... definitely WIP:
> 
> https://github.com/bollwyvl/staged-recipes/blob/add-jupyterlab-lsp/recipes/jupyterlab-lsp/meta.yaml